npm/src/compiler/compiler.ts

```diff
@@ -17,13 +17,13 @@ export interface ICompiler {
   checkCode(code: string): Promise<VSDiagnostic[]>;
   getHir(code: string): Promise<string>;
   run(
-    code: string,
+    sources: [string, string][],
     expr: string,
     shots: number,
     eventHandler: IQscEventTarget,
   ): Promise<void>;
-  getQir(code: string): Promise<string>;
-  getEstimates(code: string, params: string): Promise<string>;
+  getQir(sources: [string, string][]): Promise<string>;
```

Comment:
We'll need to coordinate this with consumers, as this is a breaking API change. Is it worth exploring alternatives? As much as it would bloat the interface, we could leave the existing APIs unchanged and have them wrap new ones that take arrays of sources. That would make it easy for consumers to switch to the project-aware APIs when they are ready and leave the behavior unchanged in the meantime.

Reply:
@minestarks, @sezna, let's circle back to this thought in a later PR.